File: src/main/java/net/Chidoziealways/everythingjapanese/chakra/IChakra.java
package net.Chidoziealways.everythingjapanese.chakra;

import net.minecraft.nbt.CompoundTag;
import net.minecraftforge.common.capabilities.AutoRegisterCapability;

@AutoRegisterCapability
public interface IChakra {
    float getChakra();
    void setChakra(float amount);
    void addChakra(float amount);
    void subtractChakra(float amount);
    int getMaxChakra();
    void setMaxChakra(int max);
    CompoundTag serializeNBT();
    void deserializeNBT(CompoundTag nbt);
    void updateMaxChakraBasedOnXP(int xpLevel);
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/chakra/IChakra.kt
package net.Chidoziealways.everythingjapanese.chakra

import net.minecraft.nbt.CompoundTag
import net.minecraft.server.level.ServerPlayer

interface IChakra {
    fun addChakra(amount: Float, player: ServerPlayer)
    fun subtractChakra(amount: Float, player: ServerPlayer) : Boolean
    fun getMaxChakra(): Int
    fun setMaxChakra(max: Int)
    fun getCurrentChakra(): Float
    fun setCurrentChakra(chakra: Float, player: ServerPlayer)
    fun setCurrentChakra(chakra: Float)
    fun serializeNBT(): CompoundTag?
    fun deserializeNBT(nbt: CompoundTag?)
    fun updateMaxChakraBasedOnXP(xpLevel: Int, player: ServerPlayer)
}
